[Index Management] Add bulk edit data retention

ElenaStoeva opened the pull request:
Closes https://github.com/elastic/kibana/issues/165142

## Summary

This PR adds bulk edit data retention to Data streams table.

On all offerings apart from serverless security project:

https://github.com/user-attachments/assets/aa091fc1-5029-4542-b76b-8045c65dda27


On serverless security project:


https://github.com/user-attachments/assets/cbccc8bd-d3f1-4511-8134-bfb0a9e8587f





### How to test:

Testing in serverless security project:
1. Start ES with `yarn es serverless --projectType security -E data_streams.lifecycle.retention.max=200d` and Kibana with `yarn serverless-security`
2. Navigate to console and create a few data streams following [these instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#create-component-templates)
3. Navigate to Index Management -> Data streams
4. Select multiple data streams and click on bulk edit data retention button
5. Verify that:
 - The help text displays the set max data retention with a link to cloud.
 - The form doesn't allow setting a data retention higher than the max data retention.
 - If any of the data streams has a reduced data retention period, an error banner is shown, displaying the affected data streams.


Testing in stateful and searc/oblt serverless project:
1. Start ES with `yarn es serverless` and Kibana with `yarn serverless-security`
2. Navigate to console and create a few data streams following [this instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#create-component-templates)
3. Navigate to Index Management -> Data streams
4. Select multiple data streams and click on bulk edit data retention button
5. Verify that:
 - There is no help text displaying the max data retention and no validation for the data retention period.
 - If any of the data streams has a reduced data retention period, an error banner is shown, displaying the affected data streams.

Configure cloud locally by adding the following to `config/kibana.dev.yml` and restarting Kibana:
```
xpack.cloud.id: 'test'
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```


### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed - https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7619




<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

ElenaStoeva committed:
[Index Management] Add bulk edit data retention

ElenaStoeva committed:
Refactor edit data retention modal

ElenaStoeva committed:
Refactor toast messages

ElenaStoeva committed:
Add help text with maximum data retention displayed

ElenaStoeva committed:
Add affected data streams callout

ElenaStoeva committed:
Only display delete button for deletable data streams

ElenaStoeva committed:
Fix type errors

ElenaStoeva committed:
Update i18n translations

ElenaStoeva committed:
Update and add api_integration tests

ElenaStoeva committed:
Fix jest tests

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/258105)
* Commit: a8946ed531e229d554ddbb1f241b3c3a52a93ce8

### Failed CI Steps
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e999-4778-9817-69714ea7484d)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e99a-452f-a350-d0a9adf9529a)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193ab04-8ac6-4457-af06-0665a66749dc)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e99c-4f11-a459-475e08eed222)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e999-4778-9817-69714ea7484d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/258105/jobs/0193aae7-e999-4778-9817-69714ea7484d/artifacts/0193ab05-7f9a-4afb-b014-12c3a973b4bd) FTR Configs #<span></span>109 / Serverless Index Management APIs Data streams Update updates the data retention of a DS
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e99a-452f-a350-d0a9adf9529a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/258105/jobs/0193aae7-e99a-452f-a350-d0a9adf9529a/artifacts/0193ab04-7ca8-4d6b-90b6-c408e7d7ba69) FTR Configs #<span></span>111 / Serverless Index Management APIs Data streams Update updates the data retention of a DS
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193aae7-e99c-4f11-a459-475e08eed222) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/258105/jobs/0193aae7-e99c-4f11-a459-475e08eed222/artifacts/0193ab10-6b38-487e-a46a-997e8280df7e) FTR Configs #<span></span>113 / Serverless Index Management APIs Data streams Update updates the data retention of a DS
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/258105#0193ab04-8ac6-4457-af06-0665a66749dc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/258105/jobs/0193ab04-8ac6-4457-af06-0665a66749dc/artifacts/0193ab14-4395-4803-9e9f-3fe7591a8560) FTR Configs #<span></span>111 / Serverless Index Management APIs Data streams Update updates the data retention of a DS

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":258105}
buildkite-pr-comment-->

SoniaSanzV reviewed (commented):
Thank you for working on this @ElenaStoeva! I tested it locally and in general looks good, but I've seen some things that I think it can be improved:

## General
The changes in the flyout have provoked that the modal width changes when modifying the retention for a single data stream: 
![data retention](https://github.com/user-attachments/assets/573ca889-a96c-4980-b4b1-3645465954ee)


When you edit the data retention for a few data streams and you first introduce a value but then select the maximum/infinite period, the data deletion warning is still there, which I believe may be confusing for users as it may seem that some data will be deleted 
<img width="723" alt="Screenshot 2024-12-10 at 08 09 31" src="https://github.com/user-attachments/assets/f46f3acb-2039-43f4-94b2-cc35081bbca7">

In the same line but not that serious, if I have a validation error and then I toggle the max retention button, the validation error is still there:
<img width="756" alt="Screenshot 2024-12-10 at 08 11 49" src="https://github.com/user-attachments/assets/9afc9896-e3d9-4a89-93fe-649ac341c15a">

Super nit pick. In the mocks the Delete data streams icon and text is red but here is all black:
<img width="525" alt="Screenshot 2024-12-10 at 08 06 56" src="https://github.com/user-attachments/assets/2215d71b-d087-444e-9281-ce869719c3df">
In the flyout the icon is red but the text is black, which I don't know if it's the expected look. 


## Security serverless
In serverless security I saw a few weird behaviors (it happens in both single edit and bulk):
* If a have some data stream set to max retention(toggle on) and then manually change the retention to the same value it says I will lose data, even thought the retention period is the same. It happens both in bulk and flyout.
* If I select a data stream and edit the retention to max (toggle on) and save, I have a toaster saying that retention is mandatory and it has been set to the maximun.

![security](https://github.com/user-attachments/assets/d754be16-e02e-4f17-a5aa-3263e6f913ae)

* When I click in the Manage project settings link, although it is an external link, it reload the page I'm working and I don't reach the Project settings page:
![Manage project settings](https://github.com/user-attachments/assets/d3142fe0-7ea7-4141-b911-14b0ad14281c)

ElenaStoeva committed:
Only show error callout when infinite data retention is switched off

ElenaStoeva committed:
Make delete button red

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259334)
* Commit: dc4a8e27439e300fab6ca269b983869b0769ac73

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0733-42b0-8a88-9c663048f3d7)
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0741-460a-bc3d-aba4dcf4b854)
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0747-4c58-820f-55d57f40916d)
* [FTR Configs #38](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-075e-4c43-af13-81bc57b45bed)
* [FTR Configs #38](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b651-8d2f-4315-bdaa-bcb852209251)
* [FTR Configs #110](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-07a6-4d49-8452-46e5ec49f83e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-075e-4c43-af13-81bc57b45bed) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-075e-4c43-af13-81bc57b45bed/artifacts/0193b651-79cc-41bf-98c6-3c1640892553) FTR Configs #<span></span>38 / Agent policies fleet_agent_policies POST /api/fleet/agent_policies should create .fleet-policies document with inputs
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b651-8d2f-4315-bdaa-bcb852209251) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b651-8d2f-4315-bdaa-bcb852209251/artifacts/0193b65a-f929-459c-88e9-e85c96baaee6) FTR Configs #<span></span>38 / Agent policies fleet_agent_policies POST /api/fleet/agent_policies should create .fleet-policies document with inputs
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0733-42b0-8a88-9c663048f3d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-0733-42b0-8a88-9c663048f3d7/artifacts/0193b658-4b3f-4d40-83a2-840160e186cc) Jest Tests #<span></span>13 / Data Streams tab data stream privileges delete displays/hides delete action depending on data streams privileges
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0733-42b0-8a88-9c663048f3d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-0733-42b0-8a88-9c663048f3d7/artifacts/0193b658-4b3f-42cf-b113-9193abf2f61a) Jest Tests #<span></span>13 / Data Streams tab when there are data streams shows all possible states according to who manages the data stream when partially managed by dsl but has backing indices managed by ILM should show a warning
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0741-460a-bc3d-aba4dcf4b854) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-0741-460a-bc3d-aba4dcf4b854/artifacts/0193b64f-8058-416a-aa60-3b816f161338) FTR Configs #<span></span>3 / Serverless Index Management APIs Data streams Update updates the data retention of a DS
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-0747-4c58-820f-55d57f40916d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-0747-4c58-820f-55d57f40916d/artifacts/0193b64b-d270-45f2-8c37-eba969db8eb7) FTR Configs #<span></span>11 / Serverless Index Management APIs Data streams Update updates the data retention of a DS
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259334#0193b62f-07a6-4d49-8452-46e5ec49f83e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259334/jobs/0193b62f-07a6-4d49-8452-46e5ec49f83e/artifacts/0193b64a-84a3-4d09-873f-fb9d2a418123) FTR Configs #<span></span>110 / Serverless Index Management APIs Data streams Update updates the data retention of a DS

### History
* :broken_heart: [Build #258105](https://buildkite.com/elastic/kibana-pull-request/builds/258105) failed a8946ed531e229d554ddbb1f241b3c3a52a93ce8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259334","number":259334,"commit":"dc4a8e27439e300fab6ca269b983869b0769ac73"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":259334}
buildkite-pr-comment-->

ryankeairns commented:
@jovana-andjelkovic would you like to review this one?

ElenaStoeva committed:
Make project settings link load in a new tab

ElenaStoeva committed:
Fix width issue

ElenaStoeva committed:
Fix logic for finding affected data streams

ElenaStoeva committed:
Fix logic for bulk or single edit

ElenaStoeva commented:
Thank you so much for the thorough testing @SoniaSanzV! I fixed most of the issues you noticed, please let me know if you still see any issues.


> The changes in the flyout have provoked that the modal width changes when modifying the retention for a single data stream

Good catch! Fixed.

> When you edit the data retention for a few data streams and you first introduce a value but then select the maximum/infinite period, the data deletion warning is still there, which I believe may be confusing for users as it may seem that some data will be deleted

Good call! Fixed.


> In the same line but not that serious, if I have a validation error and then I toggle the max retention button, the validation error is still there: 

Ah this behavior actually exists from before this PR and it's tricky to fix. I think we've discussed that on Slack before and decided that it's okay to leave it as is.

> Super nit pick. In the mocks the Delete data streams icon and text is red but here is all black: 

Nice catch! Fixed.


> ## Security serverless

> * If a have some data stream set to max retention(toggle on) and then manually change the retention to the same value it says I will lose data, even thought the retention period is the same. It happens both in bulk and flyout.

Nice catch! Fixed.

> * If I select a data stream and edit the retention to max (toggle on) and save, I have a toaster saying that retention is mandatory and it has been set to the maximum.

Fixed.

> * When I click in the Manage project settings link, although it is an external link, it reload the page I'm working and I don't reach the Project settings page:

Did you set up the cloud configs as described in the PR description? It's loading the correct page for me. I updated the link to load in a new tab though.

ElenaStoeva committed:
Revert translation changes to resolve merge conflicts

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259709)
* Commit: 1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/259709#0193baa1-977a-446c-9b07-4b25da61097d)

### History
* :broken_heart: [Build #259334](https://buildkite.com/elastic/kibana-pull-request/builds/259334) failed dc4a8e27439e300fab6ca269b983869b0769ac73
* :broken_heart: [Build #258105](https://buildkite.com/elastic/kibana-pull-request/builds/258105) failed a8946ed531e229d554ddbb1f241b3c3a52a93ce8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259709","number":259709,"commit":"1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259334","number":259334,"commit":"dc4a8e27439e300fab6ca269b983869b0769ac73"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":259709}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259763)
* Commit: 2282883cafd00bfd393ab0350b00bbb12e2e4c06

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/259763#0193baea-3f27-40ea-81d0-17c884f004bb)

### History
* :broken_heart: [Build #259709](https://buildkite.com/elastic/kibana-pull-request/builds/259709) failed 1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10
* :broken_heart: [Build #259334](https://buildkite.com/elastic/kibana-pull-request/builds/259334) failed dc4a8e27439e300fab6ca269b983869b0769ac73
* :broken_heart: [Build #258105](https://buildkite.com/elastic/kibana-pull-request/builds/258105) failed a8946ed531e229d554ddbb1f241b3c3a52a93ce8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259763","number":259763,"commit":"2282883cafd00bfd393ab0350b00bbb12e2e4c06"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259709","number":259709,"commit":"1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259334","number":259334,"commit":"dc4a8e27439e300fab6ca269b983869b0769ac73"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":259763}
buildkite-pr-comment-->

ElenaStoeva committed:
Return i18n translations back

ElenaStoeva committed:
Fix type error

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259794)
* Commit: 277838a47b39403682c89e5f165a4c9c7b4dc0b1
* [Cloud Deployment](https://kibana-pr-203083.kb.us-west2.gcp.elastic-cloud.com)

### Failed CI Steps
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/259794#0193bb23-5f42-4027-a80e-cc1950391a4e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259794#0193bb23-5f42-4027-a80e-cc1950391a4e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259794/jobs/0193bb23-5f42-4027-a80e-cc1950391a4e/artifacts/0193bb33-d458-4773-9d50-2bd81d8370d9) FTR Configs #<span></span>48 / integrations "before all" hook in "integrations"

### History
* :broken_heart: [Build #259763](https://buildkite.com/elastic/kibana-pull-request/builds/259763) failed 2282883cafd00bfd393ab0350b00bbb12e2e4c06
* :broken_heart: [Build #259709](https://buildkite.com/elastic/kibana-pull-request/builds/259709) failed 1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10
* :broken_heart: [Build #259334](https://buildkite.com/elastic/kibana-pull-request/builds/259334) failed dc4a8e27439e300fab6ca269b983869b0769ac73
* :broken_heart: [Build #258105](https://buildkite.com/elastic/kibana-pull-request/builds/258105) failed a8946ed531e229d554ddbb1f241b3c3a52a93ce8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259794","number":259794,"commit":"277838a47b39403682c89e5f165a4c9c7b4dc0b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259763","number":259763,"commit":"2282883cafd00bfd393ab0350b00bbb12e2e4c06"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259709","number":259709,"commit":"1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259334","number":259334,"commit":"dc4a8e27439e300fab6ca269b983869b0769ac73"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":259794}
buildkite-pr-comment-->

ElenaStoeva committed:
Add some fixes

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259948)
* Commit: fe45e0320575548259b5b26ea5c9c209b27e8bb4
* [Cloud Deployment](https://kibana-pr-203083.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #259794](https://buildkite.com/elastic/kibana-pull-request/builds/259794) failed 277838a47b39403682c89e5f165a4c9c7b4dc0b1
* :broken_heart: [Build #259763](https://buildkite.com/elastic/kibana-pull-request/builds/259763) failed 2282883cafd00bfd393ab0350b00bbb12e2e4c06
* :broken_heart: [Build #259709](https://buildkite.com/elastic/kibana-pull-request/builds/259709) failed 1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10
* :broken_heart: [Build #259334](https://buildkite.com/elastic/kibana-pull-request/builds/259334) failed dc4a8e27439e300fab6ca269b983869b0769ac73
* :broken_heart: [Build #258105](https://buildkite.com/elastic/kibana-pull-request/builds/258105) failed a8946ed531e229d554ddbb1f241b3c3a52a93ce8

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259948","number":259948,"commit":"fe45e0320575548259b5b26ea5c9c209b27e8bb4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259794","number":259794,"commit":"277838a47b39403682c89e5f165a4c9c7b4dc0b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259763","number":259763,"commit":"2282883cafd00bfd393ab0350b00bbb12e2e4c06"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259709","number":259709,"commit":"1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259334","number":259334,"commit":"dc4a8e27439e300fab6ca269b983869b0769ac73"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/258105","number":258105,"commit":"a8946ed531e229d554ddbb1f241b3c3a52a93ce8"}],"number":259948}
buildkite-pr-comment-->

ryankeairns reviewed (approved):
SCSS review only

SoniaSanzV commented:
Thank you for addressing my feedback @ElenaStoeva ! Everything looks good to me but one thing: In serverless security, when I change the retention to max in bulk o in the flyout I'm still seeing the warning toaster saying I didn't set a data retention period

<img width="360" alt="Screenshot 2024-12-15 at 16 35 50" src="https://github.com/user-attachments/assets/0ff9b071-b883-499b-84f1-88981ec8ba30" />

ElenaStoeva committed:
Merge branch 'main' into data-streams/bulk-edit-data-retention

ElenaStoeva commented:
> Thank you for addressing my feedback @ElenaStoeva ! Everything looks good to me but one thing: In serverless security, when I change the retention to max in bulk o in the flyout I'm still seeing the warning toaster saying I didn't set a data retention period

Hi @SoniaSanzV, thanks for re-reviewing and sorry for missing to address this point! This warning message actually comes from Elasticsearch and I believe that this behavior exists from before this PR. Our [current logic](https://github.com/elastic/kibana/blob/3a6d27af37c8ac2cdf730d206bd941554e729b9b/x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx#L272) is that if the response comes back with a warning from Es, we should display it in a toast message.

ElenaStoeva committed:
Fix i18n errors

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/260968)
* Commit: efd7ca1e708e57c80fef169e1119570fdce16468
* [Cloud Deployment](https://kibana-pr-203083.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #260831](https://buildkite.com/elastic/kibana-pull-request/builds/260831) failed 265a0b23240d07deda5c8a2ee0c2b89c30e5071f
* :broken_heart: [Build #259948](https://buildkite.com/elastic/kibana-pull-request/builds/259948) failed fe45e0320575548259b5b26ea5c9c209b27e8bb4
* :broken_heart: [Build #259794](https://buildkite.com/elastic/kibana-pull-request/builds/259794) failed 277838a47b39403682c89e5f165a4c9c7b4dc0b1
* :broken_heart: [Build #259763](https://buildkite.com/elastic/kibana-pull-request/builds/259763) failed 2282883cafd00bfd393ab0350b00bbb12e2e4c06
* :broken_heart: [Build #259709](https://buildkite.com/elastic/kibana-pull-request/builds/259709) failed 1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260968","number":260968,"commit":"efd7ca1e708e57c80fef169e1119570fdce16468"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260831","number":260831,"commit":"265a0b23240d07deda5c8a2ee0c2b89c30e5071f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259948","number":259948,"commit":"fe45e0320575548259b5b26ea5c9c209b27e8bb4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259794","number":259794,"commit":"277838a47b39403682c89e5f165a4c9c7b4dc0b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259763","number":259763,"commit":"2282883cafd00bfd393ab0350b00bbb12e2e4c06"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259709","number":259709,"commit":"1f39eb9f8dddd7bdf9a576b09bad986e28a3ed10"}],"number":260968}
buildkite-pr-comment-->

ElenaStoeva committed:
Fix jest tests

ElenaStoeva committed:
Fix serverless api integration tests

ElenaStoeva committed:
Fix type errors

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/261750)
* Commit: dc21c91141313893300c5c29f9d6481fcccf7605
* [Cloud Deployment](https://kibana-pr-203083.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #260968](https://buildkite.com/elastic/kibana-pull-request/builds/260968) failed efd7ca1e708e57c80fef169e1119570fdce16468
* :broken_heart: [Build #260831](https://buildkite.com/elastic/kibana-pull-request/builds/260831) failed 265a0b23240d07deda5c8a2ee0c2b89c30e5071f
* :broken_heart: [Build #259948](https://buildkite.com/elastic/kibana-pull-request/builds/259948) failed fe45e0320575548259b5b26ea5c9c209b27e8bb4
* :broken_heart: [Build #259794](https://buildkite.com/elastic/kibana-pull-request/builds/259794) failed 277838a47b39403682c89e5f165a4c9c7b4dc0b1
* :broken_heart: [Build #259763](https://buildkite.com/elastic/kibana-pull-request/builds/259763) failed 2282883cafd00bfd393ab0350b00bbb12e2e4c06

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261750","number":261750,"commit":"dc21c91141313893300c5c29f9d6481fcccf7605"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260968","number":260968,"commit":"efd7ca1e708e57c80fef169e1119570fdce16468"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260831","number":260831,"commit":"265a0b23240d07deda5c8a2ee0c2b89c30e5071f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259948","number":259948,"commit":"fe45e0320575548259b5b26ea5c9c209b27e8bb4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259794","number":259794,"commit":"277838a47b39403682c89e5f165a4c9c7b4dc0b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259763","number":259763,"commit":"2282883cafd00bfd393ab0350b00bbb12e2e4c06"}],"number":261750}
buildkite-pr-comment-->

SoniaSanzV reviewed (approved):
> Hi @SoniaSanzV, thanks for re-reviewing and sorry for missing to address this point! This warning message actually comes from Elasticsearch and I believe that this behavior exists from before this PR. Our [current logic](https://github.com/elastic/kibana/blob/3a6d27af37c8ac2cdf730d206bd941554e729b9b/x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx#L272) is that if the response comes back with a warning from Es, we should display it in a toast message.

Fair enough! It's not the best experience but since is not in our side, approving!

ElenaStoeva committed:
Fix jest test

ElenaStoeva committed:
Merge branch 'main' into data-streams/bulk-edit-data-retention

jovana-andjelkovic commented:
@ElenaStoeva for some reason when testing the latest build, I get some of the same issues @SoniaSanzV noted above:

- [General] When editing individual data stream retention (not through bulk action) the modal size changes due to callout
- [Security] Entering value higher than project level retention maximum, and then toggling "maximum retention" retains the error and user cannot submit the form – same happens in bulk modal as well as individual edit on data stream
![image](https://github.com/user-attachments/assets/ce736d62-c8d1-420e-8562-8e585800dc93)
![image](https://github.com/user-attachments/assets/d5b36178-cf08-439b-8f46-fe2ea58f5366)



Other issue:
- [Security] When entering different values for same selection, the callout shows different content:
    - entering `20 days` shows callout that doesn't list all affected data streams
    - entering `200 days` on same selection shows all affected data streams list in callout
    
![image](https://github.com/user-attachments/assets/aa3beefc-fafa-4bd6-86ca-1ad1f7bd4aec)
![image](https://github.com/user-attachments/assets/269867e5-c4c0-4ba1-b75b-8031298c77f7)

ElenaStoeva committed:
Fix width change and validations

ElenaStoeva commented:
Thanks for the review @jovana-andjelkovic!

> * [General] When editing individual data stream retention (not through bulk action) the modal size changes due to callout

Hmm I thought this was fixed, not sure what happened with these changes - should be fixed now with [4ebb176](https://github.com/elastic/kibana/pull/203083/commits/4ebb1763aa44d51a334ac3cb66821c35cafb1e87).
On a side note, we only want to display this error callout for non-bulk edit, correct?

> * [Security] Entering value higher than project level retention maximum, and then toggling "maximum retention" retains the error and user cannot submit the form – same happens in bulk modal as well as

I actually already addressed this in my previous comment: `Ah this behavior actually exists from before this PR and it's tricky to fix. I think we've discussed that on Slack before and decided that it's okay to leave it as is.`
I agree that this is not a great UX so I'm happy to open an issue so we can address it in a follow-up PR. Wdyt?

> * [Security] When entering different values for same selection, the callout shows different content:
>   
>   * entering `20 days` shows callout that doesn't list all affected data streams
>   * entering `200 days` on same selection shows all affected data streams list in callout

This is because for 20 days, there are 12 affected data streams and we decided that we only display the names for <= 10 data streams, right? For 200 days, there are 10 affected data streams, which is why they are displayed.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262008)
* Commit: 06012ca8512320a201d0d1df226d87211c17de5b
* [Cloud Deployment](https://kibana-pr-203083.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #261856](https://buildkite.com/elastic/kibana-pull-request/builds/261856) succeeded 1c9ecd7dd0f985ec73761b4ccb6b72e96db81c2e
* :broken_heart: [Build #261750](https://buildkite.com/elastic/kibana-pull-request/builds/261750) failed dc21c91141313893300c5c29f9d6481fcccf7605
* :broken_heart: [Build #260968](https://buildkite.com/elastic/kibana-pull-request/builds/260968) failed efd7ca1e708e57c80fef169e1119570fdce16468
* :broken_heart: [Build #260831](https://buildkite.com/elastic/kibana-pull-request/builds/260831) failed 265a0b23240d07deda5c8a2ee0c2b89c30e5071f
* :broken_heart: [Build #259948](https://buildkite.com/elastic/kibana-pull-request/builds/259948) failed fe45e0320575548259b5b26ea5c9c209b27e8bb4

cc @ElenaStoeva

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262008","number":262008,"commit":"06012ca8512320a201d0d1df226d87211c17de5b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261856","number":261856,"commit":"1c9ecd7dd0f985ec73761b4ccb6b72e96db81c2e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261750","number":261750,"commit":"dc21c91141313893300c5c29f9d6481fcccf7605"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260968","number":260968,"commit":"efd7ca1e708e57c80fef169e1119570fdce16468"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/260831","number":260831,"commit":"265a0b23240d07deda5c8a2ee0c2b89c30e5071f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259948","number":259948,"commit":"fe45e0320575548259b5b26ea5c9c209b27e8bb4"}],"number":262008}
buildkite-pr-comment-->

ElenaStoeva committed:
Add more FTRs

jovana-andjelkovic commented:
@ElenaStoeva thank you!
> Hmm I thought this was fixed, not sure what happened with these changes - should be fixed now with [4ebb176](https://github.com/elastic/kibana/pull/203083/commits/4ebb1763aa44d51a334ac3cb66821c35cafb1e87).
On a side note, we only want to display this error callout for bulk edit, correct?

It doesn't display anymore on single edit ✔️ I do think we should have it there, as well. But as it wasn't originally specified and it's not planned part of this PR – maybe we can add an issue to backlog to add it in single edit as well?

> This is because for 20 days, there are 12 affected data streams and we decided that we only display the names for <= 10 data streams, right? For 200 days, there are 10 affected data streams, which is why they are displayed.

Ah okay! In my head that limit was set on number of selected data streams rather than number of affected – but this works, maybe even better!


I think we can merge this PR! 🟢

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/12413377278

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#7619](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7619)
[✅] x-pack/test/functional/apps/index_management/config.ts: 200/200 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F203083%2Fhead)
